### PR TITLE
CTCTRADERS-568

### PR DESCRIPTION
**Pull request by kenneth-rayner**
Update simplified journey for office of presentation page. Amend Nav and Spec, show on CYA page.

### CODE REVIEW

File: app/controllers/PresentationOfficeController.scala [hunk 1]
```diff
@@ -52,24 +52,34 @@ class PresentationOfficeController @Inject()(override val messagesApi: MessagesA
   def onPageLoad(mrn: MovementReferenceNumber, mode: Mode): Action[AnyContent] =
     (identify andThen getData(mrn) andThen requireData).async {
       implicit request =>
-        request.userAnswers.get(CustomsSubPlacePage) match {
-          case Some(subsPlace) =>
+        (request.userAnswers.get(CustomsSubPlacePage), request.userAnswers.get(ConsigneeNamePage)) match {
```
Comment (chris-ianson) [inline app/controllers/PresentationOfficeController.scala:55]: If the only thing unique about the two match blocks is where the answer for presentation office comes from, you could use the match to just return the presentationOffice. Then you use this in the formProvider call and you don't have as much duplicate code

File: app/controllers/PresentationOfficeController.scala [hunk 2]
```diff
@@ -79,16 +89,16 @@ class PresentationOfficeController @Inject()(override val messagesApi: MessagesA
...
     )
     renderer.render("presentationOffice.njk", json).map(status(_))
   }
 
   def onSubmit(mrn: MovementReferenceNumber, mode: Mode): Action[AnyContent] =
     (identify andThen getData(mrn) andThen requireData).async {
       implicit request =>
-        request.userAnswers.get(CustomsSubPlacePage) match {
-          case Some(subsPlace) =>
+        (request.userAnswers.get(CustomsSubPlacePage), request.userAnswers.get(ConsigneeNamePage)) match {
```
Comment (chris-ianson) [inline app/controllers/PresentationOfficeController.scala:100]: Similar to comment above, have the match just return the presentation office and then you don't have as much duplicate code

File: conf/views/presentationOffice.njk [hunk 3]
```diff
@@ -29,7 +29,7 @@
         {% if form.errors.length > 0 %}
           {{ govukErrorSummary({
             "titleText": messages("error.summary.title"),
-            "errorList": form.errors
+            "errorList": form.errors99IT9876AB88901209
```
Comment (chris-ianson) [inline conf/views/presentationOffice.njk:32]: What's errors99IT9876AB88901209?

File: test/controllers/PresentationOfficeControllerSpec.scala [hunk 4]
```diff
@@ -48,7 +48,7 @@ class PresentationOfficeControllerSpec extends SpecBase with MockitoSugar with N
   val customsOffices            = Seq(CustomsOffice("id", "name", Seq.empty, None), CustomsOffice("officeId", "someName", Seq.empty, None))
   val form: Form[CustomsOffice] = formProvider("sub place", customsOffices)
 
-  lazy val presentationOfficeRoute: String = routes.PresentationOfficeController.onPageLoad(mrn, NormalMode).url
+  lazy val presentationOfficeRoute: String = routes.gPresentationOfficeController.onPageLoad(mrn, NormalMode).url
```
Comment (chris-ianson) [inline test/controllers/PresentationOfficeControllerSpec.scala:51]: Is there a typo here?

File: app/navigation/Navigator.scala [hunk 5]
```diff
@@ -162,10 +162,19 @@ class Navigator @Inject()() {
...
-  private def presentationOfficeRoute(mode: Mode)(ua: UserAnswers): Option[Call] =
-    (ua.get(TraderNamePage), mode) match {
+  private def consigneeAddressRoute(mode: Mode)(ua: UserAnswers): Option[Call] =
+    (ua.get(PresentationOfficePage), mode) match {
       case (Some(_), CheckMode) => Some(routes.CheckYourAnswersController.onPageLoad(ua.id))
-      case _                    => Some(routes.TraderNameController.onPageLoad(ua.id, mode))
+      case _                    => Some(routes.PresentationOfficeController.onPageLoad(ua.id, mode))
+    }
+
+  private def presentationOfficeRoute(mode: Mode)(ua: UserAnswers): Option[Call] =
```
Comment (chris-ianson) [inline app/navigation/Navigator.scala:171]: I thought the nav would get a little bloated here but seems like you have all the cases. Do you have all relevant tests for these?

File: test/controllers/PresentationOfficeControllerSpec.scala (hunk 4 above)
Comment (kenneth-rayner) [reply test/controllers/PresentationOfficeControllerSpec.scala:51]: Yeah fixed it

File: app/navigation/Navigator.scala (hunk 5 above)
Comment (kenneth-rayner) [reply app/navigation/Navigator.scala:171]: I believe so

File: test/controllers/PresentationOfficeControllerSpec.scala (hunk 4 above)
Comment (kenneth-rayner) [reply test/controllers/PresentationOfficeControllerSpec.scala]: fixed

File: conf/views/presentationOffice.njk (hunk 3 above)
Comment (kenneth-rayner) [reply conf/views/presentationOffice.njk:32]: Fixed

File: app/controllers/PresentationOfficeController.scala (hunk 2 above)
Comment (kenneth-rayner) [reply app/controllers/PresentationOfficeController.scala]: done

File: app/controllers/PresentationOfficeController.scala (hunk 1 above)
Comment (kenneth-rayner) [reply app/controllers/PresentationOfficeController.scala]: done

**Pull request merged**